polaris-patterns/src/types/i18next.d.ts

```diff
@@ -0,0 +1,11 @@
+import 'react-i18next';
+
+import type enTranslation from '../../locales/en.json';
+
+declare module 'i18next' {
+  interface CustomTypeOptions {
+    resources: {
+      translation: typeof enTranslation;
+    };
+  }
+}
```

Comment:
This fixes a [type error](https://www.i18next.com/overview/typescript#argument-of-type-defaulttfuncreturn-is-not-assignable-to-parameter-of-type-xyz) where the translate function could return `null` which isn't the same as our i18n translate function. This aligns both `i18next.t` and `i18n.translate` with the return type 'string'